apps/api/src/app/events/usecases/parse-event-request/parse-event-request.usecase.ts

```diff
@@ -286,4 +311,32 @@ export class ParseEventRequest {
 
     return reservedVariables?.map((reservedVariable) => reservedVariable.type) || [];
   }
+
+  private sanitize(subscriberId: string) {
+    if (subscriberId.trim().match(/^[a-zA-Z0-9_-]+$/)) {
+      return subscriberId;
+    }
+  }
+
+  private sanitizeRecipients(payload: TriggerRecipientsPayload): TriggerRecipientsPayload | null {
+    if (!payload) return null;
+
+    if (typeof payload === 'string') {
+      return this.sanitize(payload) || null;
+    }
+
+    if (Array.isArray(payload)) {
+      return payload
+        .map((sub) => this.sanitizeRecipients(sub as TriggerRecipientSubscriber))
```

Comment:
Do we really need recursion here? I suggest we inverse the logic to simplify it. That is, if you have one recipient wrap it with [...]. Then do [].flat() and always iterate on the array for the sanitization.

Let me know if this helps.